inmem_file_dumper.go

```diff
@@ -0,0 +1,223 @@
+package metrics
+
+import (
+	"time"
+	"sync"
+	"io/ioutil"
+	"fmt"
+	"bytes"
+	"log"
+	"sort"
+	"errors"
+)
+
+type AggregateSampleParam struct {
+	type_     AggregateSampleParamType
+	fmtString string
+}
+type AggregateSampleParamType int
+
+const (
+	Count AggregateSampleParamType = iota
+	Rate
+	Mean
+	Min
+	Max
+	Sum
+	SumSq
+	Stddev
+	LastUpdated
+)
+
+func CreateASParam(type_ AggregateSampleParamType) AggregateSampleParam {
```

Comment:
Please remove this entire file. I would recommend moving this to your own library if its something you want.